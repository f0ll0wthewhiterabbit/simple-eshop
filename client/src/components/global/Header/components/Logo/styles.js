import styled from 'styled-components'
import { Link } from 'react-router-dom'

export const Wrapper = styled.div`
  width: 100%;
  margin-bottom: ${props => props.theme.spacing(2)}px;

  ${props => props.theme.breakpoints.up('md')} {
    width: 250px;
    margin-bottom: 0;
  }
`

export const SiteLink = styled(Link)`
  margin-left: auto;
  margin-right: auto;
  width: 200px;
  display: block;
  display: flex;
  align-items: center;
  text-decoration: none;
  color: ${props => props.theme.colors.font.bold};

  ${props => props.theme.breakpoints.up('md')} {
    margin-left: 0;
    margin-right: 0;
  }
`

export const LogoImg = styled.img`
  width: 100%;
  height: auto;
  transition: filter 0.3s ease-in;

  ${props =>
    props.isDarkTheme &&
    `
      filter: brightness(4);
  `}
`
