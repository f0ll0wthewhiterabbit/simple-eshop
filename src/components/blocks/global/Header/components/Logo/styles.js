import styled from 'styled-components'
import ShopTwoRoundedIcon from '@material-ui/icons/ShopTwoRounded'
import { Typography } from '@material-ui/core'

const LogoWrapper = styled.a`
  display: block;
  flex-grow: 1;
  display: flex;
  align-items: center;
  justify-content: center;

  ${props => props.theme.breakpoints.up('lg')} {
    justify-content: flex-start;
  }
`

const LogoIcon = styled(ShopTwoRoundedIcon)`
  margin-right: ${props => props.theme.spacing(1)}px;
`

const LogoTitle = styled(Typography)`
  display: block;
  font-size: 1rem;

  ${props => props.theme.breakpoints.up('sm')} {
    display: block;
    font-size: 1.25rem;
  }
`

export { LogoIcon, LogoTitle, LogoWrapper }
