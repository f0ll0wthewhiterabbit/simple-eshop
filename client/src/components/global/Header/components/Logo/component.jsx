import React from 'react'
import PropTypes from 'prop-types'

import { MAIN_PAGE_PATH } from '../../../../../constants'
import { Wrapper, SiteLink, LogoIcon, LogoTitle } from './styles'

const Logo = ({ isAdmin }) => {
  return (
    <Wrapper data-justify={isAdmin ? 'center' : 'flex-start'}>
      <SiteLink to={MAIN_PAGE_PATH}>
        <LogoIcon fontSize="large" />
        <LogoTitle variant="h6" component="h2" data-test="LogoTitle">
          simple eShop
        </LogoTitle>
      </SiteLink>
    </Wrapper>
  )
}

Logo.propTypes = {
  isAdmin: PropTypes.bool.isRequired,
}

export default Logo
