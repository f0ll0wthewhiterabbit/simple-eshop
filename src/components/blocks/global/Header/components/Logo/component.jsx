import React from 'react'

import { LogoWrapper, LogoIcon, LogoTitle } from './styles'

const Logo = () => {
  return (
    <LogoWrapper>
      <LogoIcon fontSize="large" />
      <LogoTitle variant="h6" component="h2">
        simple eShop
      </LogoTitle>
    </LogoWrapper>
  )
}

export default Logo
