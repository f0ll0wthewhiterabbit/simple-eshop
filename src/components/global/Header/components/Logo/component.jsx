import React from 'react'

import { MAIN_PAGE_PATH } from '../../../../../constants'
import { SiteLink, LogoIcon, LogoTitle } from './styles'

const Logo = () => {
  return (
    <SiteLink to={MAIN_PAGE_PATH}>
      <LogoIcon fontSize="large" />
      <LogoTitle variant="h6" component="h2">
        simple eShop
      </LogoTitle>
    </SiteLink>
  )
}

export default Logo
