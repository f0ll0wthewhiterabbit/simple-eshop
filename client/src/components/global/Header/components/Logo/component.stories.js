import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'

import Logo from './component'

export default {
  title: 'Logo',
  component: Logo,
  decorators: [story => <Router>{story()}</Router>],
}

export const normal = () => <Logo isAdmin={false} />
export const adminMode = () => <Logo isAdmin />
