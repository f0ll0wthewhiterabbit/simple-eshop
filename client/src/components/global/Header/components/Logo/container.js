import { connect } from 'react-redux'

import Logo from './component'
import { ROLE_ADMIN, FIELD_THEME_DARK } from '../../../../../constants'

const mapStateToProps = state => ({
  isAdmin: state.getIn(['auth', 'user', 'role']) === ROLE_ADMIN,
  isDarkTheme: state.getIn(['app', 'theme']) === FIELD_THEME_DARK,
})

export default connect(mapStateToProps)(Logo)
