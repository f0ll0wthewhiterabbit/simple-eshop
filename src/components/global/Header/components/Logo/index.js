import Logo from './container'

export default Logo
