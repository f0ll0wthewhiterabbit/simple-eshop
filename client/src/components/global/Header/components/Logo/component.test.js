import React from 'react'
import { shallow } from 'enzyme'

import Logo from './component'

describe('Logo component', () => {
  let props
  let wrapper

  const generateWrapper = passedProps => {
    const initialProps = {
      isAdmin: false,
    }
    props = { ...initialProps, ...passedProps }

    return shallow(<Logo {...props} />)
  }

  beforeEach(() => {
    wrapper = generateWrapper()
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it(`should contain 'data-justify=flex-start' prop`, () => {
    expect(wrapper.prop('data-justify')).toBe('flex-start')
  })

  it(`should contain 'data-justify=center' prop in admin mode`, () => {
    wrapper = generateWrapper({ isAdmin: true })

    expect(wrapper.prop('data-justify')).toBe('center')
  })
})
