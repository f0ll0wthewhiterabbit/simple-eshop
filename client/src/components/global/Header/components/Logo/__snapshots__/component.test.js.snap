// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component should render correctly 1`] = `
<styled.div
  justify="flex-start"
>
  <Styled(Link)
    to="/"
  >
    <Styled(ExplicitIcon)
      fontSize="large"
    />
    <Styled(WithStyles(ForwardRef(Typography)))
      component="h2"
      variant="h6"
    >
      simple eShop
    </Styled(WithStyles(ForwardRef(Typography)))>
  </Styled(Link)>
</styled.div>
`;
